File: app-cpp/src/file-retrieval-engine.cpp
#include <iostream>

#include "IndexStore.hpp"
#include "ProcessingEngine.hpp"
#include "AppInterface.hpp"

int main(int argc, char** argv)
{
    int numWorkerThreads = 1;

    // TO-DO initialize the number of worker threads from argv[1]
    
    std::shared_ptr<IndexStore> store = std::make_shared<IndexStore>();
    std::shared_ptr<ProcessingEngine> engine = std::make_shared<ProcessingEngine>(store, numWorkerThreads);
    std::shared_ptr<AppInterface> interface = std::make_shared<AppInterface>(engine);

    interface->readCommands();

    return 0;
}

File: app-java/src/main/java/csc435/app/FileRetrievalEngine.java
package csc435.app;

public class FileRetrievalEngine 
{
    public static void main(String[] args)
    {
        int numWorkerThreads = 1;

        // TO-DO initialize the number of worker threads from args[0]
        
        IndexStore store = new IndexStore();
        ProcessingEngine engine = new ProcessingEngine(store, numWorkerThreads);
        AppInterface appInterface = new AppInterface(engine);

        appInterface.readCommands();
    }
}
